src/components/FeatureViews/Courses/AttendanceTable.js

```diff
@@ -0,0 +1,437 @@
+  const [attendanceBySession, setAttendanceBySession] = useState({});
+
+  useEffect(() => {
+    if(courseAttendanceMatrix.length > 0) {
+      let sessionsDoneEditing = {};
+      courseAttendanceMatrix.forEach((studentAttendanceRow) => {
+        const { attendanceList: studentAttendanceList } = studentAttendanceRow;
+        studentAttendanceList.forEach((attendanceStatus) => {
+          const { sessionId } = attendanceStatus;
+          if (attendanceStatus[sessionId] !== "UNSET") {
+            sessionsDoneEditing[sessionId] = "done";
+          };
+        });
+      });
+      setAttendanceEditStates({ ...attendanceEditStates, ...sessionsDoneEditing });
+    }
+  }, [courseAttendanceMatrix]);
+
+  const sortByFirstName = (firstStudent, secondStudent) =>
+  { 
+    if(firstStudent.studentName < secondStudent.studentName) {
+      return -1;
+    } else if(firstStudent.studentName > secondStudent.studentName) {
+      return 1;
+    } else {
+      return 0;
+    };
+  };
+
+  const { data, loading, error } = useQuery(GET_ATTENDANCE, {
+    variables: { courseId: id },
+    onCompleted: () => {
+      const { sessions, enrollments, attendances } = data;
+      setAttendanceEditStates(
+        sessions.reduce(
+          (accum, currentValue) => ({
+            ...accum,
+            [currentValue.id]: "noSelect",
+          }),
+          {}
+        )
+      );
+
+      setIsCheckBoxEditing(
+        sessions.reduce(
+          (allCheckboxStatus, {id: sessionId}) => ({ ...allCheckboxStatus, [sessionId]: false }),{})
+      );
+
+      const studentAttendanceRowsData = attendances.reduce(
+        (allStudentAttendanceRowsData, { id, enrollment, status, session }) => {
+          const studentId = enrollment.student.user.id;
+          const setInitialStudentRowData = (initalStudentRowData) =>
+            initalStudentRowData || [];
+          allStudentAttendanceRowsData[studentId] = setInitialStudentRowData(
+            allStudentAttendanceRowsData[studentId]
+          );
+          allStudentAttendanceRowsData[studentId].push({
+            attendanceId: id,
+            [session.id]: status,
+            sessionId: session.id,
+          });
+          return allStudentAttendanceRowsData;
+        },
+        {}
+      );
+      const sortBySessionId = (firstEl, secondEl) => firstEl.sessionId - secondEl.sessionId
+      const populatedCourseAttendanceMatrix = enrollments
+        .map((enrollment) =>
+          createStudentAttendanceRow(
+            fullName(enrollment.student.user),
+            studentAttendanceRowsData[enrollment.student.user.id].sort(sortBySessionId),
+            enrollment.student.user.id
+          )
+        )
+        .sort(sortByFirstName);
+      setCourseAttendanceMatrix(populatedCourseAttendanceMatrix);
+    },
+  });
+
+  const [updateAttendance] = useMutation(
+    UPDATE_ATTENDANCE_STATUS,
+    {
+      error: (err) => console.error(err),
```

Comment:
you're never updating the cache for this. would the attendance be saved if you go away from this page and return to it? (right now it persists because you have all the attendances displayed stored in state variables)